workitem/link/link_repository.go

```diff
@@ -111,6 +113,10 @@ func (r *GormWorkItemLinkRepository) Create(ctx context.Context, sourceID, targe
 		}
 		return nil, errors.NewInternalError(db.Error.Error())
 	}
+	// save a revision of the created work item link
+	if err := r.revisionRepo.Create(ctx, creatorID, RevisionTypeCreate, *link); err != nil {
+		return nil, errs.WithStack(err)
```

Comment:
don't get me wrong: I understand the purpose of wrapping, but I initially used the same message in both methods, which gave the following awkward result:
````
failed to create new work item link revision: failed to create new work item link revision: duplicate key error (or whatever SQL error you can imagine)
````

but I'll follow your recommendations and figure out something relevant ;-)